components/nexusoperations/executors.go

```diff
@@ -390,15 +375,15 @@ func (e taskExecutor) saveResult(ctx context.Context, env hsm.Environment, ref h
 }
 
 func (e taskExecutor) handleStartOperationError(env hsm.Environment, node *hsm.Node, operation Operation, callErr error) (hsm.TransitionOutput, error) {
-	var unexpectedResponseError *nexus.UnexpectedResponseError
+	var unexpectedResponseError *nexus.HandlerError
 	var opFailedError *nexus.UnsuccessfulOperationError
 
 	if errors.As(callErr, &opFailedError) {
 		return handleUnsuccessfulOperationError(node, operation, opFailedError, CompletionSourceResponse)
 	} else if errors.As(callErr, &unexpectedResponseError) {
-		if !isRetryableHTTPResponse(unexpectedResponseError.Response) {
+		if !isRetryableHTTPResponse(unexpectedResponseError.Type) {
 			// The StartOperation request got an unexpected response that is not retryable, fail the operation.
-			return handleNonRetryableStartOperationError(env, node, operation, unexpectedResponseError.Message)
+			return handleNonRetryableStartOperationError(env, node, operation, unexpectedResponseError.Failure.Message)
```

Comment:
Maybe put a comment here saying that even though `Failure` is nullable, we expect the Nexus SDK to always populate this field?